endpoints/openrtb2/auction.go

```diff
@@ -314,6 +320,113 @@ func parseTimeout(requestJson []byte, defaultTimeout time.Duration) time.Duratio
 	return defaultTimeout
 }
 
+// mergeBidderParams merges bidder parameters passed at req.ext level with imp[].ext level.
+// Preference is given to parameters at imp[].ext level over req.ext level.
+// Parameters at req.ext level are propagated to adapters as is without any validation.
+func mergeBidderParams(req *openrtb_ext.RequestWrapper) error {
+	reqBidderParams, err := adapters.ExtractReqExtBidderParams(req.BidRequest)
+	if err != nil {
+		return err
+	}
+
+	impCpy := make([]openrtb2.Imp, 0)
```

Comment:
Probably better to make this:
```suggestion
	impCpy := make([]openrtb2.Imp, len(req.BidRequest.Imp), 0)
```
This would preallocate enough space in the array under the slice for all the imps, avoiding and memory reallocations as we append to the slice.

Reply:
I believe the order here should be `len, capacity` and so it should look like:
```
impCpy := make([]openrtb2.Imp, 0, len(req.BidRequest.Imp))
```

Reply:
My mistake, thanks for the correction Mansi.

Reply:
Fixed